crates/primitives/src/transaction/mod.rs

```diff
@@ -1172,7 +1176,8 @@ impl TransactionSigned {
     /// only `true`.
     pub(crate) fn payload_len_inner(&self) -> usize {
         match &self.transaction {
-            Transaction::Legacy(legacy_tx) => legacy_tx.payload_len_with_signature(&self.signature),
+            Transaction::Legacy(legacy_tx) => legacy_tx
+                .encoded_len_with_signature(&self.signature.as_alloy_signature(legacy_tx.chain_id)),
```

Comment:
maybe worth exploring if we can phase out the reth-prim signature type?

Reply:
That would be interesting, but can it be done in a followup?